catalog/transformer/catalog-transformer-mgmp/src/main/java/org/codice/alliance/catalog/transformer/mgmp/MgmpTransformer.java

```diff
@@ -0,0 +1,604 @@
+                crsCodes.add(types.get(i) + ":" + codes.get(i));
+            }
+        }
+
+        if (CollectionUtils.isNotEmpty(mgmpTypes) && CollectionUtils.isNotEmpty(mgmpCodes)) {
+            for (int i = 0; i < mgmpCodes.size() && i < mgmpTypes.size(); i++) {
+                crsCodes.add(mgmpTypes.get(i) + ":" + mgmpCodes.get(i));
+            }
+        }
+        metacard.setAttribute(Location.COORDINATE_REFERENCE_SYSTEM_CODE, (Serializable) crsCodes);
+    }
+
+    private void addMetacardIsrInformation(final XstreamPathValueTracker pathValueTracker,
+            MetacardImpl metacard) {
+        List<String> isrComments = new ArrayList<>();
+        addValuesFromPathToListIfPresent(pathValueTracker,
+                isrComments,
+                MgmpConstants.ISR_COVERAGE_COMMENT_PATH,
+                false);
+        addValuesFromPathToListIfPresent(pathValueTracker,
+                isrComments,
+                MgmpConstants.ISR_IMAGE_COMMENT_PATH,
+                false);
+        addValuesFromPathToListIfPresent(pathValueTracker,
+                isrComments,
+                MgmpConstants.ISR_MD_IMAGE_COMMENT_PATH,
+                false);
+        addValuesFromPathToListIfPresent(pathValueTracker,
+                isrComments,
+                MgmpConstants.ISR_VIDEO_COMMENT_PATH,
+                false);
+
+        List<String> isrCategory = new ArrayList<>();
+        addValuesFromPathToListIfPresent(pathValueTracker,
+                isrCategory,
+                MgmpConstants.ISR_COVERAGE_CATEGORY_PATH,
+                false);
+        addValuesFromPathToListIfPresent(pathValueTracker,
+                isrCategory,
+                MgmpConstants.ISR_IMAGE_DESCRIPTION_PATH,
+                false);
+        addValuesFromPathToListIfPresent(pathValueTracker,
+                isrCategory,
+                MgmpConstants.ISR_MD_IMAGE_DESCRIPTION_PATH,
+                false);
+        addValuesFromPathToListIfPresent(pathValueTracker,
+                isrCategory,
+                MgmpConstants.ISR_VIDEO_DESCRIPTION_PATH,
+                false);
+
+        metacard.setAttribute(Isr.COMMENTS, (Serializable) isrComments);
+        metacard.setAttribute(Isr.CATEGORY, (Serializable) isrCategory);
+
+        addMetacardIsrOrganizationalUnit(metacard);
+    }
+
+    public void addMetacardIsrOrganizationalUnit(MetacardImpl metacard) {
+        Attribute pocName = metacard.getAttribute(Contact.POINT_OF_CONTACT_NAME);
+        Attribute publisherName = metacard.getAttribute(Contact.PUBLISHER_NAME);
+
+        if (pocName != null) {
+            setIsrOrganizationalUnitFromAttribute(pocName, metacard);
+        } else if (publisherName != null) {
+            setIsrOrganizationalUnitFromAttribute(publisherName, metacard);
+        }
+    }
+
+    private void setIsrOrganizationalUnitFromAttribute(Attribute pocName, MetacardImpl metacard) {
+        Serializable pointOfContact = pocName.getValue();
```

Comment:
An alternative (it's really a stylistic choice which to prefer) would be the streaming version:
```java
Optional.ofNullable(pocName.getValue())
        .filter(String.class::isInstance)
        .map(String.class::cast)
        .filter(StringUtils::isNotEmpty)
        .ifPresent(pocValue -> metacard.setAttribute(Isr.ORGANIZATIONAL_UNIT, pocValue));
```